drivers/mcp2515/mcp2515.h

```diff
@@ -0,0 +1,294 @@
+
+/**
+ * @brief Get MCP2515 mode of operation.
+ *
+ * @param[in]  dev          device descriptor
+ *
+ * @return                  mcp2515_mode enum
+ */
+enum mcp2515_mode mcp2515_get_mode(candev_mcp2515_t *dev);
+
+/**
+ * @brief Set MCP2515 mode of operation.
+ *
+ * @param[in]  dev          device descriptor
+ * @param[in]  mode         mode of operation to set
+ *
+ * @return                  The mode actually set
+ */
+enum mcp2515_mode mcp2515_set_mode(candev_mcp2515_t *dev,
+                                   enum mcp2515_mode mode);
+
+/**
+ * @brief Wake up MCP2515
+ *
+ * @param[in] dev           device descriptor
+ */
+void mcp2515_wake_up(candev_mcp2515_t *dev);
+
+/**
+ * @brief Get MCP2515 interrupt type.
+ *
+ * @param[in]  dev          device descriptor
+ *
+ * @return                  mcp2515_interrupt enum
+ */
+enum mcp2515_interrupt mcp2515_get_irq(candev_mcp2515_t *dev);
+
+/**
+ * @brief Clear MCP2515 interrupt.
+ *
+ * @param[in]  dev          device descriptor
+ * @param[in]  irq          interrupt to clear
+ *
+ * @return                  0 on success
+ * @return                  <0 on error
+ */
+int mcp2515_clear_irq(candev_mcp2515_t *dev, enum mcp2515_interrupt irq);
+
+/**
+ * @brief Get if an tx error occurred on MCP2515.
+ *
+ * @param[in]  dev          device descriptor
+ * @param[in]  mailbox      tx mailbox
+ *
+ * @return                  0 on success
+ * @return                  <0 on error
+ */
+int mcp2515_tx_err_occurred(candev_mcp2515_t *dev, int mailbox);
+
+/**
+ * @brief Configure the bit timing of the MCP2515.
+ *
+ * The information about the bit timing should be contained in dev descriptor.
+ *
+ * @param[in]  dev          device descriptor
+ *
+ * @return                  0 on success
+ * @return                  <0 on error
+ */
+int mcp2515_configure_bittiming(candev_mcp2515_t *dev);
+
+/**
+ * @brief Get the error flags.
+ *
+ * @param[in]  dev          device descriptor
+ *
+ * @return                  EFLG error flags
+ */
+uint8_t mcp2515_get_errors(candev_mcp2515_t *dev);
+
+/**
+ * @brief Set the @ filter_id to the position in the mailbox.
+ *
+ * @p filter_id corresponds to the position in the MCP2515 mailbox as follow:
+ *  - [0; 1]: mailbox RXB0
+ *  - [2; 5]: mailbox RXB1
+ *
+ * The MCP2515 managed 6 acceptance filters in 2 rx mailboxes:
+ *  - MB0 contains 2 acceptance filters in relation with 1 acceptance mask
+ *  - MB1 contains 4 acceptance filters in relation with 1 acceptance mask
+ *
+ *            MB0          MB1
+ *          +------+    +------+
+ *  mask 0  | RXM0 |    | RXM1 | mask 1
+ *          +======+    +======+
+ * filter 0 | RXF0 |    | RXF2 | filter 2
+ *          +------+    +------+
+ * filter 1 | RXF1 |    | RXF3 | filter 3
+ *          +------+    +------+
+ *                      | RXF4 | filter 4
+ *                      +------+
+ *                      | RXF5 | filter 5
+ *                      +------+
```

Comment:
Isn't this plot a duplicate?

Reply:
Do you mean that the plot is both in the docs for `mcp2515_set_filter()` and `mcp25155_set_mask()` ?
Yes, it seems to be like that. But do you think this is an issue? I think the plot is relevant for the description of both functions, so it makes sense to have it as a duplicate. 

But I can remove it if you think it is better to have the plot only once.

Reply:
It could be in the file header only.

Reply:
@fjmolinas Do you suggest moving it to the file header, and then referring to the plot from the function-docs?

Reply:
I though it could be addressed with having a general description of the module in the header file, and then the function header does not need to go that much into detail since the header file documents it as well. But feel free to dismiss.